linera-chain/src/manager.rs

```diff
@@ -330,6 +330,9 @@ impl ChainManager {
         // even if it is older than the current round. Validators will only sign in the current
         // round, though. (See `create_final_vote` below.)
         if let Some(locked) = &self.locked {
+            if locked.hash() == certificate.hash() {
+                return Ok(Outcome::Skip);
```

Comment:
This is an optimization, right?

Reply:
We'd also return `ChainError::InsufficientRound(_)` otherwise, because the "new" block's round isn't strictly greater, which is wrong.